### PR TITLE
yoav - fixed intake funnel

### PR DESCRIPTION
**Pull request by YoniKiriaty**

### CODE REVIEW

File: src/main/java/edu/greenblitz/robotName/subsystems/Intake/neoIntake/NeoIntake.java [hunk 1]
```diff
@@ -2,20 +2,29 @@
...
 
+import static edu.greenblitz.robotName.subsystems.Intake.neoIntake.NeoIntakeConstants.*;
+
 public class NeoIntake implements IIntake {
-	private CANSparkMax motor;
+	private GBSparkMax motor;
 	private Debouncer entranceBeamBreaker;
 	private Debouncer exitBeamBreaker;
 
 	public NeoIntake() {
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/Intake/neoIntake/NeoIntake.java:19]: spaces

File: src/main/java/edu/greenblitz/robotName/subsystems/Intake/neoIntake/NeoIntake.java [hunk 2]
```diff
@@ -2,20 +2,29 @@
...
+import edu.greenblitz.robotName.utils.motors.GBSparkMax;
 import edu.wpi.first.math.filter.Debouncer;
 
+import static edu.greenblitz.robotName.subsystems.Intake.neoIntake.NeoIntakeConstants.*;
+
 public class NeoIntake implements IIntake {
-	private CANSparkMax motor;
+	private GBSparkMax motor;
 	private Debouncer entranceBeamBreaker;
 	private Debouncer exitBeamBreaker;
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/Intake/neoIntake/NeoIntake.java:17]: spaces pls

File: src/main/java/edu/greenblitz/robotName/subsystems/Intake/IntakeConstants.java [hunk 3]
```diff
@@ -4,10 +4,7 @@
...
 public class IntakeConstants {
 
-	public static final double POWER_TO_RUN = 0;
+	public static final double POWER_TO_RUN = 0.5;
 	public static final double POWER_TO_REVERSE_RUN = -POWER_TO_RUN;
-	public static final double ENTRANCE_DEBOUNCE_TIME_FOR_LIMIT_SWITCH = 1.0;
-	public static final double EXIT_DEBOUNCE_TIME_FOR_LIMIT_SWITCH = 1.0;
-	public static final SparkMaxLimitSwitch.Type ENTRANCE_BEAM_BREAKER_TYPE = SparkMaxLimitSwitch.Type.kNormallyClosed ;
-	public static final SparkMaxLimitSwitch.Type EXIT_BEAM_BREAKER_TYPE = SparkMaxLimitSwitch.Type.kNormallyClosed ;
+
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/Intake/IntakeConstants.java:9]: spaces

**Review by YoniKiriaty — COMMENTED** [review]
i like me some spaces

File: src/main/java/edu/greenblitz/robotName/subsystems/Intake/neoIntake/NeoIntake.java [hunk 4]
```diff
@@ -2,20 +2,33 @@
...
+
 	public NeoIntake() {
-		motor = new CANSparkMax(NeoIntakeConstants.INTAKE_ID, CANSparkMaxLowLevel.MotorType.kBrushless);
+		motor = new GBSparkMax(NeoIntakeConstants.INTAKE_ID, CANSparkMaxLowLevel.MotorType.kBrushless);
+		motor.config(INTAKE_CONFIG_OBJECT);
+
 		entranceBeamBreaker = new Debouncer(NeoIntakeConstants.ENTRANCE_DEBOUNCE_TIME_FOR_LIMIT_SWITCH);
 		exitBeamBreaker = new Debouncer(NeoIntakeConstants.EXIT_DEBOUNCE_TIME_FOR_LIMIT_SWITCH);
+
+		motor.getReverseLimitSwitch(ENTRANCE_BEAM_BREAKER_TYPE).enableLimitSwitch(false);
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/Intake/neoIntake/NeoIntake.java:30]: change the 'false' to a constant

File: src/main/java/edu/greenblitz/robotName/subsystems/Intake/neoIntake/NeoIntake.java [hunk 5]
```diff
@@ -2,20 +2,33 @@
...
+		motor.getForwardLimitSwitch(EXIT_BEAM_BREAKER_TYPE).enableLimitSwitch(false);
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/Intake/neoIntake/NeoIntake.java:31]: same here

File: src/main/java/edu/greenblitz/robotName/subsystems/Intake/neoIntake/NeoIntakeConstants.java [hunk 6]
```diff
@@ -1,21 +1,31 @@
...
 
     public static final SparkMaxLimitSwitch.Type ENTRANCE_BEAM_BREAKER_TYPE = SparkMaxLimitSwitch.Type.kNormallyClosed ;
 
     public static final SparkMaxLimitSwitch.Type EXIT_BEAM_BREAKER_TYPE = SparkMaxLimitSwitch.Type.kNormallyClosed ;
 
+    public static final int CURRENT_LIMIT = 40;
+
+    public static final GBSparkMax.SparkMaxConfObject INTAKE_CONFIG_OBJECT = new GBSparkMax.SparkMaxConfObject()
+            .withIdleMode(CANSparkMax.IdleMode.kBrake)
+            .withRampRate(RobotConstants.General.RAMP_RATE_VAL)
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/Intake/neoIntake/NeoIntakeConstants.java:28]: out of CR scope, rename RAMP_RATE_VAL to RAMP_RATE_VALUE

**Review by NoamRosenberg08 — CHANGES_REQUESTED** [review]

**Review by yonnbro — APPROVED** [review]
aside from noam's comments - approved

**Review by bossyshrimp68 — APPROVED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

**Review by yonnbro — APPROVED** [review]

**Review by NoamRosenberg08 — APPROVED** [review]

**Pull request merged**